### PR TITLE
Upstreamable Morello-related changes for the purecap kernel

### PR DESCRIPTION
**Pull request by qwattash**

### CODE REVIEW

**Comment by qwattash** [comment]
> I have a change to remove it: https://reviews.freebsd.org/D30357

Pulling it into the morello branch and dropping from here

File: sys/arm64/arm64/locore.S [hunk 1]
```diff
@@ -258,7 +258,7 @@ ENTRY(mpentry)
 	/* Load the kernel page table */
 	adr	x24, pagetable_l0_ttbr1
 	/* Load the identity page table */
-	adr	x27, pagetable_l0_ttbr0_boostrap
+	adr	x27, pagetable_l0_ttbr0_bootstrap
```
Comment (brooksdavis) [inline sys/arm64/arm64/locore.S:261]: The rename should probably be it's own commit.

File: sys/kern/kern_jail.c [hunk 2]
```diff
@@ -242,7 +242,7 @@ prison0_init(void)
 			 * non-printable characters to be safe.
 			 */
 			while (size > 0 && data[size - 1] <= 0x20) {
-				data[size--] = '\0';
+				data[--size] = '\0';
```
Comment (brooksdavis) [inline sys/kern/kern_jail.c:245]: This is a really weird idiom.
